File: Practica_1/src/gestorAplicacion/personas/Domiciliario.java
package gestorAplicacion.personas;

import java.util.ArrayList;
import java.io.Serializable;

public class Domiciliario implements Serializable {

	private static final long serialVersionUID = 1L;
	private static ArrayList<Domiciliario> domiciliarios = new ArrayList<Domiciliario>();
	private String nombre;
	private int codigo;
	private boolean incidentes;


	public Domiciliario(String nombre) {
		int id = (int)(Math.random()*9000)+1000;
		this.codigo = id;
		this.nombre = nombre;
		this.incidentes = false;
		domiciliarios.add(this);
		}

	public String getNombre() {
		return nombre;
	}

	public int getCodigo() {
		return codigo;
	}

	public boolean isIncidentes() {
		return incidentes;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public void setCodigo(int codigo) {
		this.codigo = codigo;
	}

	public void setIncidentes(boolean incidentes) {
		this.incidentes = incidentes;
	}
	
	public static ArrayList<Domiciliario> getDomiciliarios() {
		return domiciliarios;
	}

	public static void setDomiciliarios(ArrayList<Domiciliario> domiciliarios) {
		Domiciliario.domiciliarios = domiciliarios;
	}

	public String toString() {
		String incidentes;
		if (this.incidentes) {
			incidentes = "este domiciliario ha sido reportado previamente";
		}else {
			incidentes = "este domiciliario no ha sido reportado previamente";
		}
		return "\nLa orden fue asignada a " + this.nombre +"\n" + incidentes;
	}
	
}

File: Practica_2/personas/domiciliario.py
import random
class Domiciliario:
    _domiciliarios = []
    
    def __init__(self, nombre):
        id = random.randint(1000, 9000)
        self._codigo = id
        self._nombre = nombre
        self._incidentes = False
        Domiciliario._domiciliarios.append(self)
    
    def getNombre(self):
        return self._nombre

    def getCodigo(self):
        return self._codigo
    
    def isIncidentes(self):
        return self._incidentes
    
    def setNombre(self, nombre):
        self._nombre = nombre
    
    def setCodigo(self, codigo):
        self._codigo = codigo
    
    def setIncidentes(self, incidentes):
        self._incidentes = incidentes
    
    @classmethod
    def getDomiciliarios(cls):
        return cls._domiciliarios

    @classmethod
    def setDomiciliarios(cls, domiciliarios):
        cls._domiciliarios = domiciliarios
    
    def __str__(self):
        incidentes = None
        
        if self._incidentes:
            incidentes = "este domiciliario ha sido reportado previamente"
        else:
            incidentes = "este domiciliario no ha sido reportado previamente"
        
        return "\nLa orden fue asignada a " + self._nombre +"\n" + incidentes
        